3.6.3 compiles, 3.8.1 fails due "optimization level attribute mismatch"

3.6.3 compiles fine, but 3.8.1 fails on the same system with

```
Making all in perl/imap
make[2]: Entering directory '/build/cyrus-imapd/src/cyrus-imapd-3.8.1/perl/imap'
Makefile:373: warning: ignoring prerequisites on suffix rule definition
Makefile:373: warning: ignoring prerequisites on suffix rule definition
cp IMAP/Admin.pm blib/lib/Cyrus/IMAP/Admin.pm
cp IMAP/Shell.pm blib/lib/Cyrus/IMAP/Shell.pm
cp IMAP.pm blib/lib/Cyrus/IMAP.pm
Running Mkbootstrap for IMAP ()
chmod 644 "IMAP.bs"
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- IMAP.bs blib/arch/auto/Cyrus/IMAP/IMAP.bs 644
"/usr/bin/perl" "/usr/share/perl5/core_perl/ExtUtils/xsubpp"  -typemap '/usr/share/perl5/core_perl/ExtUtils/typemap' -typemap '/build/cyrus-imapd/src/cyrus-imapd-3.8.1/perl/imap/typemap'  IMAP.xs > IMAP.xsc && mv IMAP.xsc IMAP.c
cc -c  -I../.. -I../../com_err/et    -I../../perl/imap -D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/perl/src=/usr/src/debug/perl -flto=auto   -DVERSION=\"1.00\" -DXS_VERSION=\"1.00\" -fPIC "-I/usr/lib/perl5/5.38/core_perl/CORE"  -DPERL_POLLUTE IMAP.c
In file included from ../../lib/util.h:49,
                 from ../../lib/imapurl.h:45,
                 from IMAP.xs:55:
../../config.h:690: warning: "VERSION" redefined
  690 | #define VERSION "3.8.1"
      | 
<command-line>: note: this is the location of the previous definition
rm -f blib/arch/auto/Cyrus/IMAP/IMAP.so
LD_RUN_PATH="/usr/lib" cc   -shared -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -flto=auto -L/usr/local/lib -fstack-protector-strong  IMAP.o  -o blib/arch/auto/Cyrus/IMAP/IMAP.so ../../perl/.libs/libcyrus.a ../../perl/.libs/libcyrus_min.a \
   -lsasl2 -lssl -lcrypto -lssl -lcrypto -luuid -lz   \
  
../../lib/util.c: In function ‘XS_Cyrus__IMAP_toURL’:
../../lib/util.c:1193:24: error: inlining failed in call to ‘always_inline’ ‘buf_len.localalias’: optimization level attribute mismatch
 1193 | EXPORTED inline size_t buf_len(const struct buf *buf)
      |                        ^
IMAP.xs:729:13: note: called from here
  729 |         if(!buf_len(&buf)) {
      |             ^
../../lib/util.c:1193:24: error: inlining failed in call to ‘always_inline’ ‘buf_len.localalias’: optimization level attribute mismatch
 1193 | EXPORTED inline size_t buf_len(const struct buf *buf)
      |                        ^
IMAP.xs:729:13: note: called from here
  729 |         if(!buf_len(&buf)) {
      |             ^
make[3]: *** [/tmp/ccD6Heye.mk:2: /tmp/ccKyRWXb.ltrans0.ltrans.o] Error 1
lto-wrapper: fatal error: make returned 2 exit status
compilation terminated.
/usr/bin/ld: error: lto-wrapper failed
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:493: blib/arch/auto/Cyrus/IMAP/IMAP.so] Error 1
make[2]: Leaving directory '/build/cyrus-imapd/src/cyrus-imapd-3.8.1/perl/imap'
make[1]: *** [Makefile:7542: all-recursive] Error 1
make[1]: Leaving directory '/build/cyrus-imapd/src/cyrus-imapd-3.8.1'
make: *** [Makefile:3348: all] Error 2
```

Any Idea how to fix that?

```
Running Mkbootstrap for IMAP ()
chmod 644 "IMAP.bs"
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- IMAP.bs blib/arch/auto/Cyrus/IMAP/IMAP.bs 644
"/usr/bin/perl" "/usr/share/perl5/core_perl/ExtUtils/xsubpp"  -typemap '/usr/share/perl5/core_perl/ExtUtils/typemap' -typemap '/build/cyrus-imapd/src/cyrus-imapd-3.8.1/perl/imap/typemap'  IMAP.xs > IMAP.xsc && mv IMAP.xsc IMAP.c
cc -c  -I../.. -I../../com_err/et    -I../../perl/imap -D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/perl/src=/usr/src/debug/perl -flto=auto   -DVERSION=\"1.00\" -DXS_VERSION=\"1.00\" -fPIC "-I/usr/lib/perl5/5.38/core_perl/CORE"  -DPERL_POLLUTE IMAP.c
In file included from ../../lib/util.h:49,
                 from ../../lib/imapurl.h:45,
                 from IMAP.xs:55:
../../config.h:690: warning: "VERSION" redefined
  690 | #define VERSION "3.8.1"
      | 
<command-line>: note: this is the location of the previous definition
rm -f blib/arch/auto/Cyrus/IMAP/IMAP.so
LD_RUN_PATH="/usr/lib" cc   -shared -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -flto=auto -L/usr/local/lib -fstack-protector-strong -fno-lto  IMAP.o  -o blib/arch/auto/Cyrus/IMAP/IMAP.so ../../perl/.libs/libcyrus.a ../../perl/.libs/libcyrus_min.a \
   -lsasl2 -lssl -lcrypto -lssl -lcrypto -luuid -lz   \
  
/usr/bin/ld: IMAP.o: plugin needed to handle lto object
/usr/bin/ld: IMAP.o: plugin needed to handle lto object
chmod 755 blib/arch/auto/Cyrus/IMAP/IMAP.so
```

although it complaints about lto objects the build succeeds.

--- a/docsrc/imap/developer/compiling.rst
+++ b/docsrc/imap/developer/compiling.rst
@@ -13,6 +13,14 @@ have a preferred distro, maybe consider using Debian.
 First make sure you have a :ref:`copy of the source <getcyrus>`. You can either
 fetch the latest source from git, or download one of our release tarballs.
 
+.. Note::
+
+    Cyrus does not support compiling with `Link Time Optimization
+    <https://gcc.gnu.org/onlinedocs/gccint/LTO-Overview.html>`_,
+    but some platforms now enable Link Time Optimization by default.
+    If your platform does so, you will need to override it, perhaps
+    by adding ``-fno-lto`` to ``CFLAGS`` and ``LDFLAGS``.
+
 Setting up dependencies
 =======================
 

